Fix cnn training notebook

jennan opened the pull request:
This PR replace the data accessor of the CNN model training notebook with a WeatherBench2 ERA5 data accessor, and fixes related issues.

jennan committed:
remove outputs of CNN notebook before refactoring

jennan committed:
move record_initialisation in subclasses, add dataset_url as property

This commit is meant to fix deserialisation issues that appeared in when training
a model using multiple processes in the CNN notebook.

jennan committed:
refactor CNN notebook (WIP) use WB2ERA5 as data source

jennan committed:
simpler implementation to pass urls, not requirement ABC

jennan committed:
fix f-strings using self before initialisation

jennan committed:
increase training period

jennan committed:
add forgotten __init__ parameters

tennlee commented:
I'm getting an exception running the notebook, let's discuss tomorrow or early next week?

jennan committed:
remove duplicate explanations of pipeline

jennan committed:
rerun CNN notebook on personal laptop with GPU

- fix comment about how to not use GPU
- increase refresh rate of rich bar to avoid warning messages

jennan committed:
remove FIXMEs and adapt comment for CUDA_VISIBLE_DEVICES to what was run

jennan commented:
@tennlee I think the PR is ready. The last changes are:

- Finished rewrite the text to remove outdated comments and some duplicated explanations,
- refreshrate of the Rich progress bar changed to avoid the notebook issues (I was able to reproduce the error),
- I have rerun it completely with a machine with a (laptop gamer) GPU.

I have not fixed the partial download issue but made an issue for this one (https://github.com/ACCESS-Community-Hub/PyEarthTools/issues/165) that I will address shortly (this week). I don't think that would block this PR.

Given that it is more convenient with a GPU, but runnable without do you still want it moved to the low-hardware requirements section?

jennan committed:
move CNN tutorial into low hardware requirement category

tennlee commented:
This has a conflict with develop/Gallery, can you please rebase and update?

jennan committed:
Merge branch 'develop' into fix_cnn_training

jennan commented:
@tennlee  sorry I am not too comfortable rebasing branches once they have been pushed to publicly accessible places, I hope you don't mind the merge instead.
